### ISSUE TITLE
ntopng on OPNsense doesn't work with HTTPS anymore

### ISSUE THREAD

**Issue by Batnun** [ntop/ntopng#6591]
Hi,

I've noticed today that ntopng doesn't work with https anymore, even when I select the certificate that always worked. I even tried to change to another certificate but no joy. The GUI is accessible only with http to port 3000.

OPNsense 22.1.7_1-amd64
FreeBSD 13.0-STABLE
OpenSSL 1.1.1o 3 May 2022
ntopng 5.3.220512

Any idea how to check/configure TLS configuration from the CLI?

**Comment by lucaderi** [comment]
can you please paste what is inside you ntopng.conf file and how you have created your certificate?

Please also paste any error in the ntopng log

**Comment by Batnun** [comment]
![image](https://user-images.githubusercontent.com/13527785/168101215-a0d4893b-5bfd-4989-9f24-afe6a89be35e.png)

**Comment by Batnun** [comment]
Can't see any related errors in the log...

**Comment by Batnun** [comment]
I'm using the same certificate that used for the OPNsense, created by XCA

**Comment by lucaderi** [comment]
In order to enable https you need to
- use -W=<port> 
and it will work

**Comment by Batnun** [comment]
> In order to enable https you need to
> 
> * use -W=
>   and it will work

You mean just replace the lowercase 'w' with uppercase 'W'? I just tried and it didn't help

See also here: https://github.com/ntop/ntopng/issues/6595

**Comment by lucaderi** [comment]
Yes. Please open the shell and start ntopng from it such as ntopng /usr/local/etc/ntopng/ntopng.conf and paste the log

**Comment by Batnun** [comment]
Just running 'ntopng' from CLI works, but with HTTP only, port 3000

/usr/local/etc/ntopng.conf is probably associated with the ntopng from the OPNsense repository, so not in use:

-i=opt1,lan,wan
-W=3000
-n=1
-d=/var/db/ntopng

/usr/local/etc/ntopng.conf is an empty file, until I start ntopng from the OPNsense GUI, and then it contain the following:

-i=igb2
-i=igb1
-i=igb0
-w=0
-W=3000
-n=1
--community=
-d=/var/db/ntopng-enterprise

However, it's already started it from the OPNsense GUI, so running from the CLI just give the following error (port in use):

![image](https://user-images.githubusercontent.com/13527785/168489532-c3351a4a-3068-405f-8d19-69dc1f7647e3.png)

I think the configuration of the ntopng on OPNsense is stored elsewhere, and being copy to this file when you enable the service from the GUI:

![image](https://user-images.githubusercontent.com/13527785/168489096-df8938fc-b993-44d3-96a3-92a77cffa072.png)

The link on the GUI to the ntopng web interface is a HTTPS to port 3000, and it doesn't work. ntopng still running HTTP only and ignore the certificate settings.

**Comment by Batnun** [comment]
Update: 

I've uninstalled ntopng from the OPNsense GUI, remove everything I've found with find / -name ntopng, and removed the repository.
Then I've installed everything according the following instructions:

https://www.ntop.org/guides/ntopng/third_party_integrations/opnsense.html

This is exactly what I did last time, but now it worked.

Thanks for the help!

**Comment by cardigliano** [comment]
Thank you. Please let us know if this happens again.

**Comment by andrewjordane2-collab** [comment]
I had a similar issue.  I had ntopng installed via OPNSense repos.  It was crashing every few hours.  

I then followed these directions https://www.ntop.org/guides/ntopng/third_party_integrations/opnsense.html to install from ntop repository.

No matter how I configured it from the OPNSense GUI, it would continue to listen on port 3000; without TLS.  

After some inspection, it turns out the configuration file wasn't being loaded in `/usr/local/etc/rc.d/ntopng`.  I updated from:

```
command=/usr/local/bin/ntopng
command_args="-U ntopng -G ${ntopng_pidfile} -1 ${httpdocs_path} -2 ${scripts_path} -3 ${callbacks_pat
h} -e"
```

to:

```
command=/usr/local/bin/ntopng
command_args="/usr/local/etc/ntopng/ntopng.conf -U ntopng -G ${ntopng_pidfile} -1 ${httpdocs_path} -2 ${scripts_path} -3 ${callbacks_pat
h} -e"
```

All works as expected now.  Just wanted to share in case it helps someone else.
